src/rest/endpoint.rs

```diff
@@ -0,0 +1,585 @@
+use std::collections::BTreeMap;
+
```

Comment:
@coderabbitai this file is too big and has a lot of messy code. Suggest way to structure this file better, split into smaller files, and add some documentation to the components defined in the file.